examples/end2end_tree_reconstruction_test.py

```diff
@@ -0,0 +1,356 @@
+    phylogeny_df = phylogeny_df.copy()
+    phylogeny_df["extant"] = phylogeny_df["taxon_label"].isin(taxon_labels)
+    phylogeny_df = alifestd_prune_extinct_lineages_asexual(
+        phylogeny_df, mutate=True
+    ).drop(columns=["extant"])
+    phylogeny_df = alifestd_collapse_unifurcations(phylogeny_df, mutate=True)
+
+    dp_tree = apc.RosettaTree(phylogeny_df).as_dendropy
+    for nd in dp_tree.preorder_node_iter():
+        nd._child_nodes.sort(
+            key=lambda nd: max(leaf.taxon.label for leaf in nd.leaf_iter()),
+        )
+    return dp_tree.as_ascii_plot()
+
+
+def load_df(path: str) -> pd.DataFrame:
+    """Helper function to get the appropriate pandas dataframe save handler."""
+    df_load_handlers = {
+        ".csv": pd.read_csv,
+        ".json": pd.read_json,
+        ".pqt": pd.read_parquet,
+        ".parquet": pd.read_parquet,
+    }
+    ext = os.path.splitext(path)[-1]
+    try:
+        return df_load_handlers[ext](path)
+    except KeyError:
+        raise ValueError(f"Unsupported dataframe path extension: {ext}")
+
+
+def sample_reference_and_reconstruction(
+    differentia_bitwidth: int,
+    surface_size: int,
+    fossil_interval: typing.Optional[int],
+) -> ReconstructionResult:
+    """Sample a reference phylogeny and corresponding reconstruction."""
+    try:
+        paths = subprocess.run(
+            [
+                f"{os.path.dirname(__file__)}/"
+                "end2end_tree_reconstruction_with_dstream_surf.sh",
+                "--differentia-bitwidth",
+                f"{differentia_bitwidth}",
+                "--surface-size",
+                f"{surface_size}",
+                *(
+                    ["--fossil-interval", f"{fossil_interval}"]
+                    * (fossil_interval is not None)
+                ),
+            ],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip()
+    except subprocess.CalledProcessError as e:
+        print(f"\033[33m{e.stdout}\033[0m")
+        print(f"\033[31m{e.stderr}\033[0m")
+        raise e
+
+    path_vars = dict()
+    exec(paths, path_vars)  # hack to load paths from shell script output
+    true_phylo_df = load_df(path_vars["true_phylo_df_path"])
+    reconst_phylo_df = alifestd_try_add_ancestor_list_col(
+        load_df(path_vars["reconst_phylo_df_path"]),
+    )  # ancestor_list column must be added to comply with alife standard
+
+    assert alifestd_count_leaf_nodes(
+        true_phylo_df
+    ) == alifestd_count_leaf_nodes(reconst_phylo_df)
+
+    reconst_phylo_df_extant = reconst_phylo_df.copy()
+    reconst_phylo_df_extant["extant"] = reconst_phylo_df["is_fossil"] == False
+    reconst_phylo_df_no_fossils = alifestd_prune_extinct_lineages_asexual(
+        reconst_phylo_df_extant
+    )
+
+    new_df = (
+        true_phylo_df.set_index("taxon_label")
+        .drop(
+            reconst_phylo_df["taxon_label"][reconst_phylo_df["is_fossil"] == True]  # type: ignore
```

Comment:
ahhh, I see. can you put this as a named variable `taxa_to_drop` right above `new_df`